Tests/BasicsTests/NetrcTests.swift

```diff
@@ -431,5 +431,65 @@ class NetrcTests: XCTestCase {
         XCTAssertEqual(netrc.machines[1].login, "fred")
         XCTAssertEqual(netrc.machines[1].password, "sunshine4ever")
     }
-}
 
+    func testComments() throws {
+        let content = """
+            # A comment at the beginning of the line
+            machine example.com # Another comment
+            login anonymous
+            password qw#erty
+            """
+
+        let netrc = try NetrcParser.parse(content)
+
+        let machine = netrc.machines.first
+        XCTAssertEqual(machine?.name, "example.com")
+        XCTAssertEqual(machine?.login, "anonymous")
+        XCTAssertEqual(machine?.password, "qw#erty")
+    }
+
+    func testHashSymbolInPassword() throws {
+        let content = """
+            machine example.com
+            login anonymous
+            password qw#erty
+            """
+
+        let netrc = try NetrcParser.parse(content)
+
+        let machine = netrc.machines.first
+        XCTAssertEqual(machine?.name, "example.com")
+        XCTAssertEqual(machine?.login, "anonymous")
+        XCTAssertEqual(machine?.password, "qw#erty")
+    }
+
+    func testQuotedPasswordWithSpace() throws {
+        let content = """
+            machine example.com
+            login anonymous
+            password "qw erty"
```

Comment:
Suggestion: in order to cover all the scenarios, can we add a flavour of that appends a comment at the end of this line

Ie
`password "we #erty" #a comment`